include/aws/io/tls_channel_handler.h

```diff
@@ -279,6 +280,57 @@ AWS_IO_API int aws_tls_ctx_options_init_client_mtls(
     const struct aws_byte_cursor *cert,
     const struct aws_byte_cursor *pkey);
 
+/**
+ * This struct exists as a graceful way to pass many arguments when
+ * calling init-with-pkcs11 functions on aws_tls_ctx_options (this also makes
+ * it easy to introduce optional arguments in the future).
+ * Instances of this struct should only exist briefly on the stack.
+ * Instructions for binding this to high-level languages:
+ * - Python: The members of this struct should be the keyword args to the init-with-pkcs11 functions.
+ * - JavaScript: This should be an options map passed to init-with-pkcs11 functions.
+ * - Java: Don't expose this struct, create function overloads with common combinations.
+ * - C++: Same as Java
+ */
+struct aws_tls_ctx_pkcs11_options {
+    /**
+     * The PKCS#11 library to use.
+     */
+    struct aws_pkcs11_lib_handle *pkcs11_lib;
+
+    /**
+     * ID of slot containing token.
+     */
+    uint32_t slot_id;
+
+    /**
+     * PIN for logging into the token (UTF-8).
+     */
+    const char *pin;
```

Comment:
the pin likely comes from user-software land like a gui prompt or a button press right? in that case, should this be an aws_string?

Reply:
This options-struct only exists for passing args on the stack. I don't intend to re-use the same struct for actual storage on the heap